manifests/social/slack.pp

```diff
@@ -0,0 +1,36 @@
+# slack.pp
+# Install Slack for OS X, Ubuntu, or Windows
+# https://www.slack.com
+#
+
+class software::social::slack (
+  $ensure  = $software::params::software_ensure,
+  $version = $software::params::slack_version,
+  $url     = $software::params::slack_url,
+) inherits software::params {
+
+  validate_string($ensure)
+
+  case $::operatingsystem {
+    'Ubuntu': {
+      $package = "slack-desktop-${version}-${::architecture}.deb"
+
+      archive { $package:
+        source  => $url,
+        path    => "/tmp/${package}",
+        extract => false,
+        cleanup => false,
+      }
+
+      package { 'slack-desktop':
```

Comment:
After you install this package, does it add an apt repo? Can you check?  The google chrome deb adds a repo. Just wondering if this one does or we can find one...

Reply:
No, it doesn't. For what I know Slack only offers a download of an installer package (for Debian, Fedora, Windows 32/64, and MacOS) off their website. I couldn't find any information about a PPA or similar.

I can ask Slack support about this to get more clarity, but for now that's what I know.

Reply:
Done. Submitted the following a support request at Slack:

> We're working on the setup automation of desktop machines, specifically on a Puppet module that allows easy rollout of the official Slack client.
> 
> Using the .deb package you provide for download off your website the installation works fine, but no Apt sources are installed for the Slack package to allow for easy automatic upgrades. Details of what we discuss are available in PR #7 of the GitHub repo we're working on:
> 
> - `https://github.com/edestecd/puppet-software/pull/7`
> 
> Do you offer or plan to offer an official repository or PPA for Debian-based installations? I didn't find any information about this online.
> 
> Kind regards,
> Peter

I hope I guessed correctly what you're targeting: Easy automatic upgrades.

Reply:
Yea its nicer to add a repo so we don't need to keep updating the version/url in params. I'm trying to get away from that. It much easier to set up the apt repo like this: https://github.com/edestecd/puppet-software/blob/master/manifests/browsers/chrome.pp#L36